src/zenml/integrations/vertex/orchestrators/vertex_orchestrator.py

```diff
@@ -0,0 +1,373 @@
+
+    from zenml.pipelines.base_pipeline import BasePipeline
+    from zenml.runtime_configuration import RuntimeConfiguration
+    from zenml.stack import Stack
+    from zenml.steps import BaseStep
+
+
+logger = get_logger(__name__)
+
+
+def _clean_pipeline_name(pipeline_name: str) -> str:
+    """Clean pipeline name to be a valid Vertex AI Pipeline name.
+
+    Arguments:
+        pipeline_name: pipeline name to be cleaned.
+
+    Returns:
+        Cleaned pipeline name.
+    """
+    return pipeline_name.replace("_", "-").lower()
+
+
+class VertexOrchestrator(BaseOrchestrator):
+    """Orchestrator responsible for running pipelines on Vertex AI.
+
+    Attributes:
+        custom_docker_base_image_name: Name of the Docker image that should be
+            used as the base for the image that will be used to execute each of
+            the steps. If no custom base image is given, a basic image of the active
+            ZenML version will be used. **Note**: This image needs to have ZenML
+            installed, otherwise the pipeline execution will fail. For that reason,
+            might want to extend the ZenML Docker images found here:
+            https://hub.docker.com/r/zenmldocker/zenml/
+        project: GCP project name. If `None`, the project will be inferred from
+            the environment.
+        location: Name of GCP region where the pipeline job will be executed.
+            Vertex AI Pipelines is available in the following regions:
+            https://cloud.google.com/vertex-ai/docs/general/locations#feature-availability
+        pipeline_root: a Google Cloud Storage path that will be used by the Vertex
+            AI Pipelines.
+        encryption_spec_key_name: The Cloud KMS resource identifier of the customer
+            managed encryption key used to protect the job. Has the form:
+            `projects/my-project/locations/my-region/keyRings/my-kr/cryptoKeys/my-key`.
+            The key needs to be in the same region as where the compute resource
+            is created.
+        synchronous: If `True`, running a pipeline using this orchestrator will
+            block until all steps finished running on Vertex AI Pipelines service.
+    """
+
+    custom_docker_base_image_name: Optional[str] = None
+    project: Optional[str] = None
+    location: str
+    pipeline_root: str
+    labels: Dict[str, str] = {}
+    encryption_spec_key_name: Optional[str] = None
+    synchronous: bool = False
+
+    FLAVOR: ClassVar[str] = VERTEX_ORCHESTRATOR_FLAVOR
+
+    @property
+    def validator(self) -> Optional[StackValidator]:
+        """Validates that the stack contains a container registry and that the
+        artifact store and metadata store used are not local."""
+
+        def _validate_stack_requirements(stack: "Stack") -> Tuple[bool, str]:
+            """Validates that all the stack components are not local."""
+            for stack_comp in stack.components.values():
```

Comment:
I was the one suggesting this local path, but now after reading your tests I realized that this doesn't cover local/non-local container registries. In this case we do have a `container_registry.is_local` property that I think we should include to fail when the stack contains a local registry.

Reply:
Fixed!